classes/core/PKPHtmlSanitizer.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * @file classes/core/PKPHtmlSanitizer.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPHtmlSanitizer
+ *
+ * @ingroup core
+ *
+ * @brief Wrapper on the top of Symfony's HtmlSanitizer implementation
+ *
+ */
+
+namespace PKP\core;
+
+use Illuminate\Support\Str;
+use Illuminate\Support\Collection;
+use PKP\config\Config;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizer;
+use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;
+use Symfony\Component\HtmlSanitizer\Reference\W3CReference;
+
+class PKPHtmlSanitizer
+{
+    /**
+     * Collection of valid element specify by W3C Sanitizer API
+     * 
+     * @see const HEAD_ELEMENTS and BODY_ELEMENTS at \Symfony\Component\HtmlSanitizer\Reference\W3CReference
+     * @see https://wicg.github.io/sanitizer-api/#default-configuration
+     */
+    protected Collection $w3cValidElements;
+
+    /**
+     * Instance of HtmlSanitizerConfig config
+     */
+    protected HtmlSanitizerConfig $htmlSanitizerConfig;
+
+    /**
+     * Create a new instance
+     */
+    public function __construct(string $allowable)
+    {
+        $this->w3cValidElements = collect(
+            array_merge(W3CReference::HEAD_ELEMENTS, W3CReference::BODY_ELEMENTS)
+        )->keys();
+
+        $this->buildSanitizerConfig(
+            $this->generateAllowedTagToAttributeMap(
+                Config::getVar('security', $allowable, null) ?? $allowable
+            )
+        );
+    }
+
+    /**
+     * Set a pre defined instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function setSanitizerConfig(HtmlSanitizerConfig $htmlSanitizerConfig): self
+    {
+        $this->htmlSanitizerConfig = $htmlSanitizerConfig;
+
+        return $this;
+    }
+
+    /**
+     * Get the instance of \Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig
+     */
+    public function getSanitizerConfig(): HtmlSanitizerConfig
+    {
+        return $this->htmlSanitizerConfig;
+    }
+
+    /**
+     * Sanitize the given html string
+     */
+    public function sanitize(string $html): string
+    {   
+        return (new HtmlSanitizer($this->htmlSanitizerConfig))->sanitize(
```

Comment:
@touhidurabir, is the cost of the sanitizer in building the configuration (cached) or instantiating the sanitizer (not cached)? If the `HtmlSanitizer` is intended to be a reusable object, would it make sense to store it as a property in this class?

Reply:
The main cost is the configure the sanitizer config as on each call of `allowElement/blockElement/...` , it clone the current object . The instantiating of the `HtmlSanitizer` only take the config object and a parser or set the default one of `Symfony\Component\HtmlSanitizer\Parser\MastermindsParser` with empty constructor . Core sanitization happens in the sanitize/For methods. But considering your suggestion, I see no harm to have it set as properly and done it .